src/Components/Register/CustomerRegistration.js

```diff
@@ -0,0 +1,160 @@
+import React, {Component} from "react";
+
+
+export default class CustomerRegistration extends Component {
+
+    state = {
+        firstName: '',
+        lastName: '',
+        Username: '',
+        password: '',
+        password2: '',
+        creditCard: '',
+        creditCardsList:[],
+    }
+
+
+
+    handleChange = event => {
+        this.setState({[event.target.name]: event.target.value});
+    }
+
+    register() {
+        //Checking to make sure all of the fields are filled out correctly
+        console.log(this.state);
+        if(this.state.firstName === '' || 
+        this.state.lastname === '' || 
+        this.state.Username === '' || 
+        this.state.password === '' || 
+        this.state.creditCard === '') {
+            window.alert("Please fill out all of the fields");
+        } else if(this.state.firstName.length > 128) {
+            window.alert("First name is too long");
+        } else if(this.state.lastName.length > 128) {
+            window.alert("Last name is too long");
+        } else if(this.state.Username.length > 128) {
+            window.alert("Username is too long");
+        } else if(this.state.password.length > 128) {
+            window.alert("Password is too long");
+        } else if(this.state.password.length < 8) {
+            window.alert("Password must be at least 8 characters long");
+        } else if(this.state.creditCard.length < 10) {
+            window.alert("Invalid credit card");
+        } else if(this.state.password !== this.state.password2) {
+            window.alert("Passwords do not match");
+        } else {
+            //Credit card validation
+            var creditString = this.state.creditCard;
+            var count = 0;
+            var cards = []
+            while(creditString && count < 5) {
+                //omit spaces and commas between credit card numbers
+                while(creditString[0] === "," || creditString[0] === " ") {
+                    creditString = creditString.substring(1);
+                }
+                var cardNumber = creditString.substring(0,10);
+
+                //if the credit card has any characters besides just numbers
+                if(cardNumber.match(/^[0-9]+$/) == null) {
+                    window.alert("Invalid credit card format");
+                    return;
+                }
+                cards[count] = cardNumber;
+                count++;
+
+                //if the user tries to add more than 5 credit cards
+                if(count >= 5) {
```

Comment:
does this show the alert on the 5th card?

Reply:
You're right, it should just be `count > 5`